### PR TITLE
feat: add oidc auth

### LINKED ISSUE #3: add login command
1. bc-cli login命令成功后， 生成config文件 `~/.bestchains/config`

```yaml
apiVersion: v1beta1
cluster:
    server: https://172.22.96.133:6443
     name: xxx
user:
  name: xxx
  idtoken: `xxxxx`
saas:
  depository:
      auth: xxx
      server: https://xxxxx:xxx
```

2. 后续所有的CLI调用的参数解析按照如下流程：

1） 首先读取config文件，获取本地配置
2） 读取CLI flags，覆盖本地配置

3. 如果`.auth` 为 `oidc`，则需要将`idtoken`加载到http request header中

### PR DESCRIPTION
**Pull request by Abirdcfly**
Fix #3

### CODE REVIEW

**Comment by Abirdcfly** [comment]
see https://github.com/bestchains/bc-cli/issues/3#issuecomment-1537057796

File: pkg/auth/auth.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
+package auth
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
```
Comment (bjwswang) [inline pkg/auth/auth.go:7]: run make verify before commit

**Review by bjwswang — APPROVED** [review]
LGTM but remeber to  do `make veriy` before push this pr

**Review by bjwswang — APPROVED** [review]
LGTM but remeber to  do `make veriy` before push this pr

**Review by 0xff-dev — APPROVED** [review]

**Pull request merged**